abcclassroom/tests/test_assignment_template.py

```diff
@@ -85,11 +85,9 @@ def test_create_template_dir_fail_when_exists(default_config, tmp_path):
     default_config["course_directory"] = tmp_path
     assignment_name = "test_assignment"
     abctemplate.create_template_dir(default_config, assignment_name)
-    expected_path = str(
-        Path(default_config["template_dir"], assignment_name + "-template")
-    )
+
     # If run again, it should fail given the dir already exists
-    with pytest.raises(Exception, match=("Directory " + expected_path)):
+    with pytest.raises(Exception, match=("Oops! The directory specified")):
```

Comment:
Here i was matching the text + the path and it  caused an issue only on windows. no need to do that given we just want to make sure our error and our associated message is actually being presented to the user!

Reply:
We should also raise a `FileExistsError` in `template.py`, and then write this test as `pytest.raises(FileExistsError, match=("Oops! The directory specified"))`. Thinking ahead to using this as a library, and throwing an exception specific to the type of error that we encounter for future clients.

Reply:
totally. i hadn't dug into that yet because i was confused by the fail mode and how / when it's triggered. so i was going to look at that next. i was also confused by why delete mode keeps the .git directory! if you could explain that it would be helpful! once i understand the modes better it will be easier to add checks!

Reply:
The `fail` mode is the default for the `abc-new-template` script. That mode is not available for `abc-update-template`. As for the `.git` dir, this is tricky. There is some explanation at https://github.com/earthlab/abc-classroom/blob/master/abcclassroom/template.py#L169. Can explain more, but its complicated and this is not the best venue!